protocol/messenger.go

```diff
@@ -4138,6 +4138,31 @@ func (m *Messenger) DeleteMessagesByChatID(id string) error {
 	return m.persistence.DeleteMessagesByChatID(id)
 }
 
+func (m *Messenger) MarkMessageAsUnreadImpl(chatID string, messageID string) (uint64, uint64, error) {
+	count, countWithMentions, err := m.persistence.MarkMessageAsUnread(chatID, messageID)
+
+	if err != nil {
+		return 0, 0, err
+	}
+
+	chat, err := m.persistence.Chat(chatID)
+	if err != nil {
+		return 0, 0, err
+	}
+	m.allChats.Store(chatID, chat)
+	return count, countWithMentions, nil
+}
+
+func (m *Messenger) MarkMessageAsUnread(chatID string, messageID string) (uint64, uint64, error) {
+	count, countWithMentions, err := m.MarkMessageAsUnreadImpl(chatID, messageID)
+
+	if err != nil {
+		return 0, 0, err
+	}
+
```

Comment:
i think it's right to also check activity center here. If persistence has AC notification with `id == messageID` it also should be marked as unseen. Basically same as here: https://github.com/status-im/status-go/blob/develop/protocol/messenger.go#L4173

Reply:
> i think it's right to also check activity center here. If persistence has AC notification with `id == messageID` it also should be marked as unseen. Basically same as here: https://github.com/status-im/status-go/blob/develop/protocol/messenger.go#L4173

Let me have a look.